### ISSUE TITLE
You cannot add type safe listener on link

### ISSUE THREAD

**Issue by tomitrescak** [projectstorm/react-diagrams#148]
Link uses inherited "addListener" that uses "BaseModelListener" as parameter, while it should extend this method with addListener(listener: LinkModelListener). Otherwise you receive type errors:

the error is in the followinf .d.ts file. It should be:

```typescript
export declare class BaseModel<T extends BaseModelListener = BaseModelListener> extends BaseEntity<T> 
```

instead of 

```typescript
export declare class BaseModel<T extends BaseModelListener = BaseModelListener> extends BaseEntity<BaseModelListener>
```

**Comment by dylanvorster** [comment]
I think this is fixed in the links refactor PR thats currently open

**Comment by dylanvorster** [comment]
fixed in 5.0.0
